[tests-only][full-ci] extend webdav tests for old, new and spaces path

PrajwolAmatya opened the pull request:
<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the oCIS component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next version of oCIS.

Please set the following labels:

- Set label "Status:Needs-Review" for review or "Status:In-Progress" if the PR still has open tasks.
- Assignment: assign to self
- Reviewers: pick at least one
-->

## Description
This PR extends the test scenarios that use Webdav API to cover all available dav path versions (i.e. old, new and spaces).

- For chunk upload the chunk version 1 is set as default, since the chunking version 2 uses the path `remote.php/dav/uploads` which doesn't exist in oCIS. Also discussed in issue https://github.com/owncloud/ocis/issues/1321.

Build check for both configs (with and without remote.php):
- https://drone.owncloud.com/owncloud/ocis/41071

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- https://github.com/owncloud/ocis/issues/10183

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in owncloud.github.io/ -->
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

amrita-shrestha commented on tests/acceptance/features/apiGraph/unassignRole.feature:
```diff
@@ -17,11 +18,19 @@ Feature: unassign user role
-      | Space Admin |
-      | User        |
-      | User Light  |
+      | user-role   | dav-path-version |
+      | Admin       | old              |
+      | Space Admin | old              |
+      | User        | old              |
+      | User Light  | old              |
+      | Admin       | new              |
+      | Space Admin | new              |
+      | User        | new              |
+      | User Light  | new              |
+      | Admin       | spaces           |
+      | Space Admin | spaces           |
+      | User        | spaces           |
+      | User Light  | spaces           |
```
do graph api endpoint affected by dav-path-version?

saw-jan replied:
yeah, we can leave it as it is. no need to run with multiple dav paths (though we have one dav When step - the main actions is unassigning role)

PrajwolAmatya replied:
I added the `dav-path-version` because of the following steps:
```Scenario
When user "Brian" uploads file with content "this step will assign the role to default" to "assign-to-default.txt" using the WebDAV API
And user "Brian" should have the role "User" assigned
```

I will revert the changes back to original. :+1:

saw-jan commented on tests/acceptance/bootstrap/WebDav.php:
```diff
@@ -1755,22 +1753,9 @@ public function userUploadsAFileInChunk(
-		);
+
+		// use chunking version 1 as default, since version 2 uses "remote.php/dav/uploads" endpoint and it doesn't exist in oCIS
+		$this->chunkingToUse = 1;
```
we need to do the proper cleanup. Get rid of `chunkingToUse` but that would be in coming PRs

PrajwolAmatya replied:
Issue for cleanup: https://github.com/owncloud/ocis/issues/10596

saw-jan commented on tests/acceptance/features/apiAntivirus/antivirus.feature:
we might have duplicate scenarios after adding spaces dav path.
https://github.com/owncloud/ocis/blob/076c92f259981c798c68e1975bbe7742b908a289/tests/acceptance/features/apiAntivirus/antivirus.feature#L176
https://github.com/owncloud/ocis/blob/076c92f259981c798c68e1975bbe7742b908a289/tests/acceptance/features/apiAntivirus/antivirus.feature#L230
https://github.com/owncloud/ocis/blob/076c92f259981c798c68e1975bbe7742b908a289/tests/acceptance/features/apiAntivirus/antivirus.feature#L439
https://github.com/owncloud/ocis/blob/076c92f259981c798c68e1975bbe7742b908a289/tests/acceptance/features/apiAntivirus/antivirus.feature#L503

saw-jan commented on tests/acceptance/features/apiDepthInfinity/propfind.feature:
```diff
@@ -149,5 +153,6 @@ Feature: PROPFIND with depth:infinity
     Then the HTTP status code should be "400"
     Examples:
       | dav-path-version |
+      | old              |
```
there's no old trashbin endpoint
```suggestion
```

saw-jan commented on tests/acceptance/features/apiDepthInfinity/propfind.feature:
```diff
@@ -118,23 +117,28 @@ Feature: PROPFIND with depth:infinity
       | permissionsRole | view          |
     When the public lists the resources in the last created public link with depth "infinity" using the WebDAV API
     Then the HTTP status code should be "400"
+    Examples:
+      | dav-path-version |
+      | new              |
+      | spaces           |
```
we only have one public link endpoint

saw-jan commented on tests/acceptance/features/apiSharingNg1/listPermissions.feature:
```diff
@@ -167,8 +167,7 @@ Feature: List a sharing permissions
 
 
   Scenario: user lists permissions of a project space
-    Given using spaces DAV path
-    And user "Brian" has been created with default attributes and without skeleton files
```
let's keep these. because new dav path is the default one

saw-jan commented on tests/acceptance/features/coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:
```diff
@@ -505,8 +505,9 @@ Feature: sharing
   @issue-764 @issue-7555
-  Scenario: share a file by multiple channels and download from sub-folder and direct file share
-    Given these users have been created with default attributes and without skeleton files:
+  Scenario Outline: share a file by multiple channels and download from sub-folder and direct file share
+    Given using <dav-path-version> DAV path
+    And these users have been created with default attributes and without skeleton files:
```
let's keep it as it is. ocs sharing will be dead soon.

saw-jan commented on tests/acceptance/features/coreApiShareManagementToShares/mergeShare.feature:
```diff
@@ -126,10 +127,16 @@ Feature: sharing
     And the HTTP status code of responses on all endpoints should be "200"
     And as "Brian" folder "/Shares/merge-test-outside-groups-renamebeforesecondshare" should exist
     But as "Brian" folder "/merge-test-outside-groups-renamebeforesecondshare-renamed" should not exist
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+      | spaces           |
```
we can revert this one.

saw-jan commented on tests/acceptance/features/coreApiSharePublicLink1/accessToPublicLinkShare.feature:
```diff
@@ -79,3 +80,7 @@ Feature: accessing a public link share
       | password        | %public% |
     When user "Alice" tries to download file "textfile.txt" from the last public link using own basic auth and new public WebDAV API
     Then the HTTP status code should be "401"
+    Examples:
+      | dav-path-version |
+      | new              |
+      | spaces           |
```
we only have one public webdav path

saw-jan commented on tests/acceptance/features/coreApiTrashbin/trashbinDelete.feature:
```diff
@@ -26,6 +26,8 @@ Feature: files and folders can be deleted from the trashbin
     And as "Alice" the file with original path "<file-name-2>" should not exist in the trashbin
     Examples:
       | dav-path-version | file-name     | file-name-2   |
+      | old              | textfile0.txt | textfile1.txt |
+      | old              | sample,0.txt  | sample,1.txt  |
```
we do not have old trashbin path

saw-jan commented on tests/acceptance/features/coreApiTrashbin/trashbinFilesFolders.feature:
```diff
@@ -17,6 +17,7 @@ Feature: files and folders exist in the trashbin after being deleted
     But as "Alice" file "/textfile0.txt" should not exist
     Examples:
       | dav-path-version |
+      | old              |
```
we do not have old trashbin path

saw-jan commented on tests/acceptance/features/coreApiTrashbin/trashbinSharingToShares.feature:
```diff
@@ -27,6 +27,7 @@ Feature: using trashbin together with sharing
     And as "Brian" the folder with original path "/Shares/renamed_shared" should not exist in the trashbin
     Examples:
       | dav-path-version |
+      | old              |
```
we do not have old trashbin path

PrajwolAmatya replied:
These scenarios uses following when step:
```feature
When user "Brian" deletes folder "/Shares/renamed_shared" using the WedDAV API
```
So, using all dav paths.

saw-jan commented on tests/acceptance/features/coreApiTrashbinRestore/trashbinRestore.feature:
```diff
@@ -142,6 +142,7 @@ Feature: restore deleted files/folders
       | /textfile0.txt |
     Examples:
       | dav-path-version | http-status-code |
+      | old              | 404              |
```
we do not have old trashbin path

saw-jan commented on tests/acceptance/features/coreApiVersions/disableFileVersioning.feature:
```diff
@@ -9,31 +9,43 @@ Feature: checking file versions
     When user "Alice" gets the number of versions of file "/testfile.txt"
     Then the HTTP status code should be "207"
     And the number of versions should be "0"
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+      | spaces           |
```
I think, we only have one endpoint for file versions.

saw-jan commented on tests/acceptance/features/coreApiVersions/fileVersions.feature:
```diff
@@ -5,129 +5,208 @@ Feature: dav-versions
 
 
-  Scenario: upload file and no version is available
+  Scenario Outline: upload file and no version is available
+    Given using <dav-path-version> DAV path
```
This is fine. When step is a upload step

saw-jan reviewed (approved):
LGTM :+1:

PrajwolAmatya commented:
After addressing reviews, some line numbers have changed so running CI with and without `remote.php` in dav path. Only merge after all the pipelines are green.
Build check: https://drone.owncloud.com/owncloud/ocis/41176
